main/sdr.go

```diff
@@ -319,21 +314,164 @@ func sdrKill() {
+	//         UAT    ES    id 1 -> ES  second pass loop 1, id 0 -> Unconfigured
+	//         ES     ES    id 0 -> ES  first  pass loop 1, id 1 -> unconfigured
+	//         UAT    UAT   id 0 -> Unconfigured          , id 1 -> Unconfigured
+
+	// es_enabled == false and uat_enabled == true
+	//
+	// ID      0      1
+	// -------------------------------------------------------------------------
+	//         A      A     id 0 -> UAT first  pass loop 2, id 1 -> Unconfigured
+	//         A      ES	id 0 -> UAT first  pass loop 2, id 1 -> Unconfigured
+	//         ES     A     id 1 -> UAT second pass loop 2, id 0 -> Unconfigured
+	//         A      UAT   id 1 -> UAT second pass loop 1, id 0 -> Unconfigured
+	//         UAT    A     id 0 -> UAT first  pass loop 1, id 1 -> Unconfigured
+	//         ES     UAT   id 1 -> UAT second pass loop 1, id 0 -> Unconfigured
+	//         UAT    ES    id 0 -> UAT first  pass loop 1, id 1 -> Unconfigured
+	//         ES     ES    id 0 -> Unconfigured          , id 1 -> Unconfigured
+	//         UAT    UAT   id 0 -> UAT first  pass loop 1, id 1 -> unconfigured
+
+	// es_enabled == false and uat_enabled == false
+	//
+	// ID      0      1
+	// -------------------------------------------------------------------------
+	//         A      A     id 0 -> Unconfigured          , id 1 -> Unconfigured
+	//         A      ES	id 0 -> Unconfigured          , id 1 -> Unconfigured
+	//         ES     A     id 0 -> Unconfigured          , id 1 -> Unconfigured
+	//         A      UAT   id 0 -> Unconfigured          , id 1 -> Unconfigured
+	//         UAT    A     id 0 -> Unconfigured          , id 1 -> Unconfigured
+	//         ES     UAT   id 0 -> Unconfigured          , id 1 -> Unconfigured
+	//         UAT    ES    id 0 -> Unconfigured          , id 1 -> Unconfigured
+	//         ES     ES    id 0 -> Unconfigured          , id 1 -> Unconfigured
+	//         UAT    UAT   id 0 -> Unconfigured          , id 1 -> Unconfigured
+
+	// once the tagged dongles have been assigned explicitly range over
+	// the remaining IDs and assign then to any anonymous dongles
+	unusedIDs := make([]int, 0, count)
+
+	// loop 1: assign tagged dongles
+	for i := 0; i < count; i++ {
+		_, _, s, err := rtl.GetDeviceUsbStrings(i)
+		if err == nil {
+			if uat_enabled && UATDev == nil && rUAT.hasID(s) {
+				createUATDev(i, s, true)
+			} else if es_enabled && ESDev == nil && rES.hasID(s) {
+				createESDev(i, s, true)
+			} else {
+				unusedIDs = append(unusedIDs, i)
+			}
+		} else {
+			log.Printf("rtl.GetDeviceUsbStrings id %d: %s\n", i, err)
+		}
 	}
 
+	// loop 2; assign anonymous dongles
+	for _, v := range unusedIDs {
+		_, _, s, err := rtl.GetDeviceUsbStrings(v)
+		if err == nil {
+			if uat_enabled && UATDev == nil {
+				createUATDev(v, s, false)
+			} else if es_enabled && ESDev == nil {
+				createESDev(v, s, false)
+			}
+		} else {
+			log.Printf("rtl.GetDeviceUsbStrings id %d: %s\n", v, err)
+		}
+	}
+}
```

Comment:
Looks great but I think your big comment block is incorrect; since loop 2 doesn't check, for example,
`if uat_enabled && UATDev == nil && !rES.hasID(s)`
loop 2 will automatically override (ignore?) any dual-assignments. In my opinion, ideal behavior would include such a check in loop 2 and then override that check in a loop 3, but the existing code is very close to ideal behavior. A case in which this might arise is if someone connected 4 SDRs: one 1090 connected to a ES antenna, two 978s connected to UAT antennas, and one anonymous connected to a generically-lengthed antenna, and the 1090 SDR fails; the existing code might assign the second 978 to ES when the correct choice may be the anonymous one.

But the code is already ideal for my setup.